vowpalwabbit/plt.cc

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#include <cfloat>
+#include <cmath>
+#include <cstdio>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <algorithm>
+#include <queue>
+#include "reductions.h"
+#include "vw.h"
+
+using namespace VW::LEARNER;
+using namespace VW::config;
+
+namespace plt_ns
+{
+struct node
+{
+  uint32_t n;  // node number
+  float p;     // node probability
+
+  bool operator<(const node &r) const { return p < r.p; }
+};
+
+struct plt
+{
+  vw *all;
```

Comment:
Nit: pointer should be with type.

```suggestion
  vw* all;
```

Reply:
Fixed, note: clang-format does not fix it since `DerivePointerAlignment: true` option.

Reply:
Thanks for letting me know!